src/source/vector_tile_worker_source.ts

```diff
@@ -66,8 +67,18 @@ export class VectorTileWorkerSource implements WorkerSource {
      */
     async loadVectorTile(params: WorkerTileParameters, abortController: AbortController): Promise<LoadVectorTileResult> {
         const response = await getArrayBuffer(params.request, abortController);
+        const metadataResponse = await getArrayBuffer({
+            url: params.metadataRequest.url,
+        }, new AbortController());
+
         try {
-            const vectorTile = new vt.VectorTile(new Protobuf(response.data));
+            const metadata = new Uint8Array(metadataResponse.data);
+            const tilesetMetadata = TileSetMetadata.fromBinary(metadata);
+
+            const vectorTile = new vt.VectorTile(
+                new Uint8Array(response.data),
```

Comment:
This `new Uint8Array` allocation may be expensive. We should investigate why this seems needed and how to avoid this so we can pass the `response.data` directly. Perhaps create a ticket on this and I can take a closer look?

Reply:
I'm down to open an issue for this, the root of it is that the decoder needs a `Uint8Array` but we're getting an `ArrayBuffer` here. It seems like `Uint8Array` is the common type between node and the browser here?

Reply:
A similar conversion may be happening in pbf [here](https://github.com/mapbox/pbf/blob/main/index.js#L8).